sklearn/decomposition/nmf.py

```diff
@@ -44,7 +44,7 @@ def trace_dot(X, Y):
 
 
 def _check_init(A, shape, whom):
-    A = check_array(A)
+    A = check_array(A, variable_name='A')
```

Comment:
the variable name is actually included in `whom` (see calls to the function, below). Replace `whom` with `variable_name`.